Pcf 15

kjwinfield opened the pull request:
### Summary
**STAR aligner**
Sentieon's STAR aligner takes multiple fastq.gz files for one sample as inputs. It aligns these to a reference genome and outputs a .bam and .bam.bai file of the alignment. It also outputs a chimeric.out.junction file of chimeric reads, which may indicate gene fusions. Every line in this output file contains one chimerically aligned read.

This DNAnexus app runs STAR aligner.

Links to STAR and Sentieon STAR documentation:

- [Sentieon RNA calling](https://support.sentieon.com/versions/202112.05/manual/RNA_call/rna/)
- [STAR manual]( https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf)

**Examples**

**Pre-review**
Example DNAnexus job to test: [job-GGVQp3Q42QQq931K0xbxPbff](https://platform.dnanexus.com/projects/GFfV0qQ42QQVf6VX12zxxvx3/monitor/job/GGVQp3Q42QQq931K0xbxPbff)
Inputs and outputs:
![Screenshot from 2022-09-15 13-07-31](https://user-images.githubusercontent.com/71272357/190399803-4a2da8ff-0797-4178-ac55-24db0c4bec0a.png)
Screenshot showing chimeric.out.junction file:
![Screenshot from 2022-09-15 13-08-33](https://user-images.githubusercontent.com/71272357/190400693-f4de38bb-5fce-4a5c-b080-25b41dcf2325.png)

**Post-review**
After this pull request was opened changes were made including:
- The app now takes multiple R1 and R2 files as inputs to the aligner without concatenating them first
- Generating the read group identifier for the bam has been automated

Example DNAnexus job using the new version: [job-GGqKZP042QQX84G86z9z0X7j](https://platform.dnanexus.com/projects/GFfV0qQ42QQVf6VX12zxxvx3/monitor/job/GGqKZP042QQX84G86z9z0X7j) 
Inputs and outputs:
![Screenshot from 2022-09-30 16-24-35](https://user-images.githubusercontent.com/71272357/193303992-7948e183-dbdd-4670-aaf2-d9562fcdd7eb.png)
Screenshot showing chimeric.out.junction file:
![Screenshot from 2022-09-30 16-25-23](https://user-images.githubusercontent.com/71272357/193303983-5342033b-dbd8-4c08-a936-526eca82ff9c.png)


<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)
<!-- Reviewable:end -->

kjwinfield committed:
got sentieon license working

kjwinfield committed:
attempting to call sentieon tools in script

kjwinfield committed:
added sentieon tools to path

kjwinfield committed:
added sentieon star command + set defaults

kjwinfield committed:
corrected output command

kjwinfield committed:
Update README.md

kjwinfield committed:
outputs bam and bai files

kjwinfield committed:
outputs bam and bai files

kjwinfield committed:
outputs bam and bai files

kjwinfield committed:
outputs bam and bai files

kjwinfield committed:
chimeric junction output created

kjwinfield committed:
added concatenating files

kjwinfield committed:
wrote some tests

kjwinfield committed:
make tests work

kjwinfield committed:
corrected typos

kjwinfield committed:
tidied up dxapp json

kjwinfield committed:
tidied up dxapp json

kjwinfield committed:
added comments

kjwinfield committed:
edited readme, removed output of concatenated R1 and R2 files

kjwinfield committed:
edited readme

kjwinfield committed:
edit readme

kjwinfield committed:
edit readme

corbin-chris reviewed (changes requested):
> Reviewed 4 of 8 files at r1, 5 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 2 unresolved discussions (waiting on @kjwinfield and @mattgarner)

___
*[`src/script.sh` line 65 at r3](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC4h2WO3-qw3jCt_Y6B:-NC4h2WO3-qw3jCt_Y6C:b-n7p5kz) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/3176236340aef689546b8a3eea8fe0a7be6bf183/src/script.sh#L65)):*
> ```Shell
>   R2_test[$i]=${R2_test[$i]//$to_cut_R2/};
>   R2_test[$i]=${R2_test[$i]//$cut_fastq/}
> done
> ```

I'd normally advise against code reuse and advise writing functions instead. An untested example is given below. However I'm not sure it's compatible with how this is set-up, as there isn't a main() method for control flow, so I'd run it past Matt first.



_Code snippet:_
```Shell
# Check R1 and R2 are paired correctly; for each R1 there should be a matching R2
# Create test arrays that are equal to the arrays for R1 and R2
R1_test=${R1[@]}
R2_test=${R2[@]}

trim_fastq_endings () {
  # Takes array of fastq files with their read number ("R1" or "R2"), trims the endings off every file, and returns as an array
  fastq_array=$1
  read_to_cut=$2
  cut_fastq=".fastq.gz"
  for i in "${!fastq_array[@]}"; do
    fastq_array[$i]=${fastq_array[$i]//$read_to_cut/};
    fastq_array[$i]=${fastq_array[$i]//$cut_fastq/};
  done
  echo $fastq_array
}

R1_test=$(trim_fastq_endings $R1_test "R1")
R2_test=$(trim_fastq_endings $R2_test "R2")
```
___
*[`src/script.sh` line 111 at r3](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC4w730-5CAwQORHE5f:-NC4w730-5CAwQORHE5g:b-5ch06s) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/3176236340aef689546b8a3eea8fe0a7be6bf183/src/script.sh#L111)):*
> ```Shell
>     --chimOutJunctionFormat 1 \
>     | sentieon util sort -r ${REFERENCE} -o ${SORTED_BAM} -t ${NUMBER_THREADS} -i -
> ```

Splitting every arg onto its own line would be easier to read


<!-- Sent from Reviewable.io -->

kjwinfield committed:
split command onto multiple lines

mattgarner reviewed (changes requested):
> Reviewed 4 of 8 files at r1, 2 of 5 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 11 unresolved discussions (waiting on @kjwinfield)

___
*[`src/script.sh` line 7 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC57x5-AlTQF285p_DR:-NC57x5-AlTQF285p_DS:bsvux6c) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L7)):*
> ```Shell
> dx-download-all-inputs # download inputs from json
> 
> # resources folder will not exist on worker, so removed here.
> ```

I think was added for your benefit when figuring out dx apps, so I would suggest removing this now.



_Code quote:_
```Shell
# resources folder will not exist on worker, so removed here.
```
___
*[`src/script.sh` line 33 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5DHduABLg_X-G1sA0:-NC5DHduABLg_X-G1sA1:b-10rahw) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L33)):*
> ```Shell
> export PATH="$SENTIEON_BIN_DIR:$PATH"
> 
> # Concatenate the R1 and R2 files over multiple lanes into one file
> ```

See section 3.2 of https://raw.githubusercontent.com/alexdobin/STAR/master/doc/STARmanual.pdf

Do we need to concat them?



_Code quote:_
```Shell
Concatenate the R1 and R2 files over multiple lanes into one file
```
___
*[`src/script.sh` line 91 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5BFJoEMqgQDeGQDZE:-NC5BFJoEMqgQDeGQDZF:b-dhytgj) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L91)):*
> ```Shell
> 
> # Run STAR-aligner
> NUMBER_THREADS=32
> ```

According to the STAR manual this needs to be set to the number of cores available.

```
--runThreadN option defines the number of threads to be used for genome generation, it has
to be set to the number of available cores on the server node
```
Source: https://raw.githubusercontent.com/alexdobin/STAR/master/doc/STARmanual.pdf

The app is set to use the mem1_ssd1_x32 instance type, so this is fine. Until someone runs a job which runs out of resources and fails, so they change the instance type and you no longer have 32 cores.

Perhaps you could get and use the number of threads available in this particular instance, and set this value dynamically based on that. IF you do this, don't forget to check that different instance types generate the same output for the same input.



_Code quote:_
```Shell
32
```
___
*[`src/script.sh` line 96 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5CvDOEA2owe1OruCp:-NC5CvDOEA2owe1OruCq:bazx1bq) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L96)):*
> ```Shell
> SAMPLE=/home/dnanexus/R1/${sample_name}_R1_concat.fastq.gz
> SAMPLE2=/home/dnanexus/R2/${sample_name}_R2_concat.fastq.gz
> GROUP_NAME=${read_group}
> ```

Let's think about dealing with read groups properly. 

Some reading below, then let's talk
https://gatk.broadinstitute.org/hc/en-us/articles/360035890671-Read-groups



_Code quote:_
```Shell
${read_group}
```
___
*[`src/script.sh` line 99 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5E6ZfEzRd-5VUOA0F:-NC5E6ZfEzRd-5VUOA0G:b-3gbvvd) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L99)):*
> ```Shell
> SAMPLE_NAME=${sample_name}
> PLATFORM=ILLUMINA
> READ_LENGTH_MINUS_1=100
> ```

If we think a fixed value here is appropriate could we perhaps link to something to justify this in a comment?



_Code quote:_
```Shell
100
```
___
*[`src/script.sh` line 100 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5EWVN9Uz6Gk95y2XY:-NC5EWVN9Uz6Gk95y2XZ:bj3lwvg) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L100)):*
> ```Shell
> PLATFORM=ILLUMINA
> READ_LENGTH_MINUS_1=100
> SORTED_BAM="/home/dnanexus/out/${sample_name}.bam"
> ```

Consider including something in the bam filename which identifies this as a star aligned bam.



_Code quote:_
```Shell
.bam
```
___
*[`src/script.sh` line 105 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5F94EBnFz_25sVUP_:-NC5F94EBnFz_25sVUPa:b-1kogpp) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L105)):*
> ```Shell
>     --genomeDir ${STAR_REFERENCE} \
>     --readFilesIn ${SAMPLE} ${SAMPLE2} \
>     --readFilesCommand "zcat" \
> ```

Anything except a .gz compressed fastq file will fail, so either extract the extension from the input files and adjust this value as appropriate, or enforce that only .gz files are accepted as input via the dxapp.json



_Code quote:_
```Shell
zcat
```
___
*[`src/script.sh` line 107 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5FbIKDXXI1bWM3azr:-NC5FbIKDXXI1bWM3azs:bqbi8ja) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L107)):*
> ```Shell
>     --readFilesCommand "zcat" \
>     --outStd BAM_Unsorted \
>     --outSAMtype BAM Unsorted \
> ```

Could we sort our output?



_Code quote:_
```Shell
    --outStd BAM_Unsorted \
    --outSAMtype BAM Unsorted \
```
___
*[`src/script.sh` line 113 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5FrqV4Gb9WXEGYzWC:-NC5FrqV4Gb9WXEGYzWD:bvs6jlq) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L113)):*
> ```Shell
>     --twopass1readsN -1 \
>     --sjdbOverhang ${READ_LENGTH_MINUS_1} \
>     --chimSegmentMin 12 \
> ```

Have these params been reviews/justified?



_Code quote:_
```Shell
    --twopassMode Basic \
    --twopass1readsN -1 \
    --sjdbOverhang ${READ_LENGTH_MINUS_1} \
    --chimSegmentMin 12 \
    --chimJunctionOverhangMin 8 \
    --chimOutJunctionFormat 1 \
```


<!-- Sent from Reviewable.io -->

mattgarner reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 11 unresolved discussions (waiting on @corbin-chris and @kjwinfield)

___
*[`src/script.sh` line 65 at r3](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC4h2WO3-qw3jCt_Y6B:-NC5G62c6_sBCeR_m0UC:b-bbqotn) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/3176236340aef689546b8a3eea8fe0a7be6bf183/src/script.sh#L65)):*
<details><summary><i>Previously, corbin-chris (Chris Corbin) wrote…</i></summary><blockquote>

I'd normally advise against code reuse and advise writing functions instead. An untested example is given below. However I'm not sure it's compatible with how this is set-up, as there isn't a main() method for control flow, so I'd run it past Matt first.
</blockquote></details>

Matt agrees, good suggestion :+1:


<!-- Sent from Reviewable.io -->

corbin-chris reviewed (commented):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 10 unresolved discussions (waiting on @kjwinfield)



<!-- Sent from Reviewable.io -->

kjwinfield committed:
added function, instance type input, ability to take multiple files as input

kjwinfield committed:
added comment

kjwinfield committed:
attempting to add read group info

kjwinfield committed:
add comment

kjwinfield committed:
add comment

kjwinfield committed:
added readgroup information, sentieon STAR command now takes manifest as input for read group info

kjwinfield committed:
manifest now includes platform and sample info in @RG line

kjwinfield committed:
typo

kjwinfield reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 7 of 9 files reviewed, 10 unresolved discussions (waiting on @corbin-chris, @kjwinfield, and @mattgarner)

___
*[`src/script.sh` line 65 at r3](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC4h2WO3-qw3jCt_Y6B:-NCeQMziBdbC8S0Sm6R7:bb6un1j) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/3176236340aef689546b8a3eea8fe0a7be6bf183/src/script.sh#L65)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Matt agrees, good suggestion :+1:
</blockquote></details>

Done
___
*[`src/script.sh` line 111 at r3](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC4w730-5CAwQORHE5f:-NC52_SL0sgXoEZkHHEP:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/3176236340aef689546b8a3eea8fe0a7be6bf183/src/script.sh#L111)):*
<details><summary><i>Previously, corbin-chris (Chris Corbin) wrote…</i></summary><blockquote>

Splitting every arg onto its own line would be easier to read
</blockquote></details>

Done.
___
*[`src/script.sh` line 7 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC57x5-AlTQF285p_DR:-NCewSH02ZJdX_Wq3K-W:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L7)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

I think was added for your benefit when figuring out dx apps, so I would suggest removing this now.
</blockquote></details>

Done.
___
*[`src/script.sh` line 91 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5BFJoEMqgQDeGQDZE:-NCeQWcgAneX4Q-6O_5q:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L91)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

According to the STAR manual this needs to be set to the number of cores available.

```
--runThreadN option defines the number of threads to be used for genome generation, it has
to be set to the number of available cores on the server node
```
Source: https://raw.githubusercontent.com/alexdobin/STAR/master/doc/STARmanual.pdf

The app is set to use the mem1_ssd1_x32 instance type, so this is fine. Until someone runs a job which runs out of resources and fails, so they change the instance type and you no longer have 32 cores.

Perhaps you could get and use the number of threads available in this particular instance, and set this value dynamically based on that. IF you do this, don't forget to check that different instance types generate the same output for the same input.
</blockquote></details>

Done.
___
*[`src/script.sh` line 96 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5CvDOEA2owe1OruCp:-NDDfcL68NZq6BEKs5IT:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L96)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Let's think about dealing with read groups properly. 

Some reading below, then let's talk
https://gatk.broadinstitute.org/hc/en-us/articles/360035890671-Read-groups
</blockquote></details>

Done.
___
*[`src/script.sh` line 99 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5E6ZfEzRd-5VUOA0F:-ND3q_6SBWdoEB6YmeZu:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L99)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

If we think a fixed value here is appropriate could we perhaps link to something to justify this in a comment?
</blockquote></details>

Done.
___
*[`src/script.sh` line 100 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5EWVN9Uz6Gk95y2XY:-NCdwPx74V42EdfWKuRO:bs7wumb) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L100)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Consider including something in the bam filename which identifies this as a star aligned bam.
</blockquote></details>

Done.
___
*[`src/script.sh` line 105 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5F94EBnFz_25sVUP_:-NCdxH_C--knqixTkwpr:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L105)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Anything except a .gz compressed fastq file will fail, so either extract the extension from the input files and adjust this value as appropriate, or enforce that only .gz files are accepted as input via the dxapp.json
</blockquote></details>

Done.
___
*[`src/script.sh` line 107 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5FbIKDXXI1bWM3azr:-NC5IIk81GE68AUAUoPC:b5qdvpj) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L107)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Could we sort our output?
</blockquote></details>

does this command sort it?



_Code quote:_
```Shell
| sentieon util sort -r ${REFERENCE} -o ${SORTED_BAM} -t ${NUMBER_THREADS} -i -
```
___
*[`src/script.sh` line 113 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5FrqV4Gb9WXEGYzWC:-NCdzgyg2tq1ZEP8D-IG:b4m139o) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L113)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Have these params been reviews/justified?
</blockquote></details>

Parameters from STAR-fusion wiki: https://github.com/STAR-Fusion/STAR-Fusion/wiki


<!-- Sent from Reviewable.io -->

kjwinfield committed:
update readme

kjwinfield committed:
now includes sample name in chimeric.out.junctions file

kjwinfield committed:
edit delimiter in file name

kjwinfield committed:
try to fix libjemalloc error

kjwinfield reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 6 of 9 files reviewed, 10 unresolved discussions (waiting on @corbin-chris and @mattgarner)

___
*[`src/script.sh` line 33 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5DHduABLg_X-G1sA0:-NC5HPJs7mo1OxU2hPLQ:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L33)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

See section 3.2 of https://raw.githubusercontent.com/alexdobin/STAR/master/doc/STARmanual.pdf

Do we need to concat them?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

kjwinfield committed:
set default read length as 100

kjwinfield committed:
fixed error with libjemalloc

kjwinfield committed:
made defaults no chr versions

mattgarner reviewed (commented):
> Reviewed 1 of 1 files at r7, 1 of 3 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 7 of 9 files reviewed, 4 unresolved discussions (waiting on @corbin-chris and @kjwinfield)

___
*[`README.md` line 19 at r7](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NDSFBpW5SHe-5D_VQyM:-NDSFBpXCJjLwDtUr-0X:b6jjrvt) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/aac5a8cde138ccacc144b894bd17b69db8f235d9/README.md#L19)):*
> ```Markdown
> 
> ## Notes
> * This app is not ready for production use
> ```

This will need to be updated before we make a v1.0.0



_Code quote:_
```Markdown
This app is not ready for production use
```


<!-- Sent from Reviewable.io -->

kjwinfield committed:
configured app to output log files

kjwinfield committed:
updated readme to include log files as output

kjwinfield committed:
fix typo

kjwinfield reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 6 of 9 files reviewed, 4 unresolved discussions (waiting on @corbin-chris and @mattgarner)

___
*[`README.md` line 19 at r7](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NDSFBpW5SHe-5D_VQyM:-NFheh515zL9FFVJabpM:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/aac5a8cde138ccacc144b894bd17b69db8f235d9/README.md#L19)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

This will need to be updated before we make a v1.0.0
</blockquote></details>

Done.
___
*[`src/script.sh` line 113 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5FrqV4Gb9WXEGYzWC:-NFhemGvCK_RfDncwBK5:b-7kkjq1) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L113)):*
<details><summary><i>Previously, kjwinfield (Katherine Winfield) wrote…</i></summary><blockquote>

Parameters from STAR-fusion wiki: https://github.com/STAR-Fusion/STAR-Fusion/wiki
</blockquote></details>

We can also adjust these later depending on the downstream apps i.e. STAR-Fusion and FusionInspector


<!-- Sent from Reviewable.io -->

kjwinfield committed:
remove version info

kjwinfield committed:
now handles both fastq.gz and fq.gz suffixes

kjwinfield committed:
change variable name to make more indicative

mattgarner reviewed (commented):
> Reviewed 1 of 3 files at r10.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 7 of 9 files reviewed, 3 unresolved discussions (waiting on @corbin-chris)



<!-- Sent from Reviewable.io -->

mattgarner reviewed (changes requested):
> Reviewed 2 of 3 files at r10, 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 3 unresolved discussions (waiting on @corbin-chris and @kjwinfield)

___
*[`src/script.sh` line 113 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5FrqV4Gb9WXEGYzWC:-NGpKTnB6zjlkVe4Ua1F:bnq4k4) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L113)):*
<details><summary><i>Previously, kjwinfield (Katherine Winfield) wrote…</i></summary><blockquote>

We can also adjust these later depending on the downstream apps i.e. STAR-Fusion and FusionInspector
</blockquote></details>

If we expect to want to adjust these params then should they be inputs?


<!-- Sent from Reviewable.io -->

kjwinfield committed:
reconfigured to use CTAT library

kjwinfield commented:
### New version that uses CTAT library
CTAT provenance: on [Confluence](https://cuhbioinformatics.atlassian.net/l/cp/uz1tfet3)
DNAnexus job: [job-GJkqKK842QQZ3b1jFpQ6Pf25](https://platform.dnanexus.com/projects/GFfV0qQ42QQVf6VX12zxxvx3/monitor/job/GJkqKK842QQZ3b1jFpQ6Pf25)
![Screenshot from 2022-11-15 16-49-07](https://user-images.githubusercontent.com/71272357/201979230-a7fa1046-6f5a-49aa-ac2b-4a3e7bdb5db9.png)
![Screenshot from 2022-11-15 16-55-34](https://user-images.githubusercontent.com/71272357/201979682-be5d5fdb-0e58-4b17-994c-20e9c562d5a3.png)

kjwinfield reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 6 of 9 files reviewed, 3 unresolved discussions (waiting on @corbin-chris and @mattgarner)

___
*[`src/script.sh` line 99 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5E6ZfEzRd-5VUOA0F:-NGwDc-i84pP2CFyLQJW:b-t2vbwi) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L99)):*
<details><summary><i>Previously, kjwinfield (Katherine Winfield) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Read length is now an input and there is a comment explaining


<!-- Sent from Reviewable.io -->

corbin-chris reviewed (commented):
It's looking good, only 1 comment

> Reviewed 3 of 3 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-:-NH-2h8I5WhRxoetzcxI:b-gt7k2a)* status: all files reviewed, 4 unresolved discussions (waiting on @kjwinfield and @mattgarner)

___
*[`src/script.sh` line 22 at r12](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NGzxe4BDUpbUbP6zlDB:-NGzxe4BDUpbUbP6zlDC:bj9pikp) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/500eaf0554488fb87f375efa0cdcfcdf5aba0389/src/script.sh#L22)):*
> ```Shell
> # Move genome indices and reference genome to specific folders
> 
> mv /home/dnanexus/genome_lib/*.plug-n-play/ctat_genome_lib_build_dir/ref_genome.fa.star.idx/* /home/dnanexus/genomeDir/
> ```

For future-proofing it is worth searching for the CTAT using:
/home/dnanexus/genome_lib/*CTAT_lib*/ctat_genome_lib_build_dir/ref_genome.fa.star.idx/*
instead of searching on 'plug-n-play. This is because 'plug-n-play' indicates the pre-built libs by the CTAT people, and if we ever get custom-built CTAT libraries working, I'd not want to name it 'plug-n-play' to avoid confusion. eggd_fusioninspector searches on the string "CTAT_lib" and eggd_starfusion uses tonnes of wildcards to find CTAT and barely names anything.
@mattgarner we can let this slide for now in the name of speed if needs be


<!-- Sent from Reviewable.io -->

corbin-chris reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 3 unresolved discussions (waiting on @kjwinfield and @mattgarner)



<!-- Sent from Reviewable.io -->

kjwinfield committed:
reconfigured to take parameters as input

kjwinfield committed:
edit readme

kjwinfield commented:
### Parameters are now an input string
![Screenshot from 2022-11-16 11-09-52](https://user-images.githubusercontent.com/71272357/202200898-d369ff30-be5f-42aa-af08-f7ec7ba1432f.png)

kjwinfield committed:
code now extracts CTAT lib directory name from input

kjwinfield reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 6 of 9 files reviewed, 2 unresolved discussions (waiting on @corbin-chris and @mattgarner)

___
*[`src/script.sh` line 113 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5FrqV4Gb9WXEGYzWC:-NH0AXyGDsPdhbtIxRZ7:b-vfra4t) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L113)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

If we expect to want to adjust these params then should they be inputs?
</blockquote></details>

They are now inputs. Additional parameters are input as a string.


<!-- Sent from Reviewable.io -->

mattgarner reviewed (changes requested):
> Reviewed 2 of 3 files at r13, 1 of 1 files at r14, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 3 unresolved discussions (waiting on @kjwinfield)

___
*[`README.md` line 11 at r14](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NH8cobO59hSdrk2dAiD:-NH8cobP-w3bKFeL1a6E:b566k2p) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/bf3753907a2fd8b14f7db879dfaf4e4f0c36cf58/README.md#L11)):*
> ```Markdown
> * `--read_length`: (int) The length of the genomic sequence around the annotated junction to be used in constructing the splice junctions database. Ideally, this length should be equal to the ReadLength-1, where ReadLength is the length of the reads. The reference genome and genome indices from the CTAT library needs this to be set at 150, so that is the default.
> * `--genome_lib`:(file) A CTAT genome library, which is a reference file bundle required by Trinity CTAT tools. This contains the genome indices and reference genome that are needed to run Sentieon STAR aligner
> * `--parameters`: (string) The parameters with which to run Sentieon's STAR aligner. Should be space-delimated, and in the format `--parameter-name value` e.g. `--alignIntronMax 100000 --alignSJstitchMismatchNmax 5 -1 5 5`. Has default parameters set if not configured by the user.
> ```

Are there any restrictions on what can be passed via this string? I assume you wouldn't want --readFilesIn, for example



_Code quote:_
```Markdown
--parameters
```
___
*[`src/script.sh` line 99 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5E6ZfEzRd-5VUOA0F:-NGpGU049YRN6X_L4iXI:b-k5mly3) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L99)):*
<details><summary><i>Previously, kjwinfield (Katherine Winfield) wrote…</i></summary><blockquote>

Read length is now an input and there is a comment explaining
</blockquote></details>

Would it be better to ask the user to specify the actual readlength, then subtract one within the code rather than relying on the user to remember to do this themselves?

I think it would be quite easy for the user to get this wrong, especially given that the input is called `read_length`, which could reasonably be taken to mean length of the reads, not length of the reads minus 1
___
*[`src/script.sh` line 160 at r14](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NH8dOipB66_LIUHjJpn:-NH8dOipB66_LIUHjJpo:b-1ag9k3) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/bf3753907a2fd8b14f7db879dfaf4e4f0c36cf58/src/script.sh#L160)):*
> ```Shell
>     --readFilesIn ${SAMPLE} ${SAMPLE2} \
>     --readFilesCommand "zcat" \
>     --readFilesManifest "/home/dnanexus/fastqs/manifest.tsv" \
> ```

Is it correct to provide both 
`--readFilesIn`
and 
`--readFilesManifest`

It seems like the manifest makes the first option redundant



_Code quote:_
```Shell
    --readFilesIn ${SAMPLE} ${SAMPLE2} \
    --readFilesCommand "zcat" \



    --readFilesManifest "/home/dnanexus/fastqs/manifest.tsv" \
```


<!-- Sent from Reviewable.io -->

kjwinfield committed:
edit readme

kjwinfield committed:
edit readme

kjwinfield committed:
edit readme

kjwinfield committed:
edit readme

kjwinfield reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 8 of 9 files reviewed, 3 unresolved discussions (waiting on @mattgarner)

___
*[`README.md` line 11 at r14](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NH8cobO59hSdrk2dAiD:-NH9H0ae5qbJyjbR7mr7:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/bf3753907a2fd8b14f7db879dfaf4e4f0c36cf58/README.md#L11)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Are there any restrictions on what can be passed via this string? I assume you wouldn't want --readFilesIn, for example
</blockquote></details>

Done.
___
*[`src/script.sh` line 160 at r14](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NH8dOipB66_LIUHjJpn:-NH9O54q4fUlmR2v0Bzh:bitd5sm) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/bf3753907a2fd8b14f7db879dfaf4e4f0c36cf58/src/script.sh#L160)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Is it correct to provide both 
`--readFilesIn`
and 
`--readFilesManifest`

It seems like the manifest makes the first option redundant
</blockquote></details>

this is not the case. the manifest just specifies read group info, --readFilesIn is still needed to provide the input files
e.g. this job is the same app, but has --readFilesIn removed and did not work: https://platform.dnanexus.com/projects/GFfV0qQ42QQVf6VX12zxxvx3/monitor/job/GJqbZZj42QQkygZx30ZfYJVQ


<!-- Sent from Reviewable.io -->

kjwinfield committed:
rename read_length input to clarify its purpose

kjwinfield reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 6 of 9 files reviewed, 3 unresolved discussions (waiting on @mattgarner)

___
*[`src/script.sh` line 99 at r4](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NC5E6ZfEzRd-5VUOA0F:-NHANxCXD2GNJTCwQRBN:b-qbpapt) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/37855bf045ad75ba4f961756932e734f5aa0b4b0/src/script.sh#L99)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Would it be better to ask the user to specify the actual readlength, then subtract one within the code rather than relying on the user to remember to do this themselves?

I think it would be quite easy for the user to get this wrong, especially given that the input is called `read_length`, which could reasonably be taken to mean length of the reads, not length of the reads minus 1
</blockquote></details>

edited to clarify that this is the read length used in generation of the CTAT genome library, in this case 151 bp
the app subtracts 1 to make the input for readlength-1 in STAR 150


<!-- Sent from Reviewable.io -->

kjwinfield committed:
typo

kjwinfield committed:
correct typo in readme

kjwinfield committed:
correct typo in readme

kjwinfield committed:
adding markduplicates into the read me, reconfigured app so readgroups work

kjwinfield commented:
The eggd_staraligner app has been edited:

- Add the new output `${sample_name}.mark_duplicates.star.Processed.out.bam `, which has duplicate reads marked
- Make read groups work by using the STAR input `--outSAMattrRGline`, rather than `--readFilesManifest`

This dnanexus job: [job-GKJXX3j42QQxXXz7BBYF75KG](https://platform.dnanexus.com/projects/GFfV0qQ42QQVf6VX12zxxvx3/monitor/job/GKJXX3j42QQxXXz7BBYF75KG) uses the new version and demonstrates these features

mattgarner reviewed (changes requested):
> Reviewed 3 of 3 files at r17, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 6 unresolved discussions (waiting on @kjwinfield)

___
*[`README.md` line 11 at r14](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NH8cobO59hSdrk2dAiD:-NJe2lU88dcKOxirfU7Q:b-2a99el) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/bf3753907a2fd8b14f7db879dfaf4e4f0c36cf58/README.md#L11)):*
<details><summary><i>Previously, kjwinfield (Katherine Winfield) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Does the app check for the presence of these within the string which is passed in?
___
*[`src/script.sh` line 22 at r12](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NGzxe4BDUpbUbP6zlDB:-NJe36sh-UfOpB5Fe5uZ:bb6un1j) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/500eaf0554488fb87f375efa0cdcfcdf5aba0389/src/script.sh#L22)):*
<details><summary><i>Previously, corbin-chris (Chris Corbin) wrote…</i></summary><blockquote>

For future-proofing it is worth searching for the CTAT using:
/home/dnanexus/genome_lib/*CTAT_lib*/ctat_genome_lib_build_dir/ref_genome.fa.star.idx/*
instead of searching on 'plug-n-play. This is because 'plug-n-play' indicates the pre-built libs by the CTAT people, and if we ever get custom-built CTAT libraries working, I'd not want to name it 'plug-n-play' to avoid confusion. eggd_fusioninspector searches on the string "CTAT_lib" and eggd_starfusion uses tonnes of wildcards to find CTAT and barely names anything.
@mattgarner we can let this slide for now in the name of speed if needs be
</blockquote></details>

Done
___
*[`src/script.sh` line 133 at r17](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NJeGdwN5O3Kku57gQnt:-NJeGdwO149u6PIzIK8e:b-op0ry6) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/6a587cbd6817a58810eeb509c20fede32ef7c572/src/script.sh#L133)):*
> ```Shell
> for L in ${fq_arr[@]}; do echo $(_check_for_string ${L}) >> file.txt; done
> 
> unique_read_groups=$(for L in ${fq_arr[@]}; do echo $(_check_for_string ${L}); done)
> ```

This is not used after being defined. Remove?



_Code quote:_
```Shell
unique_read_groups
```
___
*[`src/script.sh` line 154 at r17](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NJeGwde8yzpMtc5GbrO:-NJeGwde8yzpMtc5GbrP:b-knjp30) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/6a587cbd6817a58810eeb509c20fede32ef7c572/src/script.sh#L154)):*
> ```Shell
> read_group_list=$(while IFS=' ' read -r R1 R2; do
>     lane=$( echo $R1 | cut -d'_' -f3)  # The lanes are not necessarily L001 and L002, so this grabs the lane information from the fastq file name
>     rg=$(grep $lane manifest.tsv | cut -f3-)  # Extract the read group
> ```

Let's make this grep for _L001_ rather than L001 - makes it less likely to pick up the wrong thing
(e.g. a sample named `XL00123` would cause the current matching to fail)



_Code quote:_
```Shell
rg=$(grep $lane manifest.tsv | cut -f3-)  # Extract the read group
```
___
*[`src/script.sh` line 160 at r17](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NJeHi9yDUcXqz0sd8X7:-NJeHi9yDUcXqz0sd8X8:bh27no9) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/6a587cbd6817a58810eeb509c20fede32ef7c572/src/script.sh#L160)):*
> ```Shell
> # Split the read groups into first and last, so that they can be passed to the --outSAMattrRGline input to the aligner
> first_rg=$(echo "$read_group_list" | cut -d$'\n' -f1)
> last_rg=$(echo "$read_group_list" | cut -d$'\n' -f2)
> ```

This will fail if we have != 2 RGs. A more generalised approach would be:

First, the reads for each RG need to be passed to `--readFilesIn`  in Lane sorted order within each R1/R2 group
like:
`X1_S1_R1_L001.fastq.gz,X1_S1_R1_L002.fastq.gz,X1_S1_R1_L003.fastq.gz X1_S1_R2_L001.fastq.gz,X1_S1_R2_L002.fastq.gz,X1_S1_R2_L003.fastq.gz`
or simplified
`L001_R1,L002_R1,L003_R1 L001_R2,L002_R2,L003_R2`

then all read groups need to be passed to `--outSAMattrRGline` in the same Lane sorted order:
`L001 , L002 , L003`



_Code quote:_
```Shell
first_rg=$(echo "$read_group_list" | cut -d$'\n' -f1)
last_rg=$(echo "$read_group_list" | cut -d$'\n' -f2)
```


<!-- Sent from Reviewable.io -->

mattgarner reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 6 unresolved discussions (waiting on @kjwinfield)

___
*[`src/script.sh` line 154 at r17](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NJeGwde8yzpMtc5GbrO:-NJoWtrBBJpSF66YRryM:b-v4pccj) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/6a587cbd6817a58810eeb509c20fede32ef7c572/src/script.sh#L154)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Let's make this grep for _L001_ rather than L001 - makes it less likely to pick up the wrong thing
(e.g. a sample named `XL00123` would cause the current matching to fail)
</blockquote></details>

Markdown broke my comment. Should _L001_ should be \_L001\_ 


<!-- Sent from Reviewable.io -->

mattgarner reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 6 unresolved discussions (waiting on @kjwinfield)

___
*[`src/script.sh` line 154 at r17](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NJeGwde8yzpMtc5GbrO:-NJoX7On6RgNnUHgZcEh:bzgfvv2) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/6a587cbd6817a58810eeb509c20fede32ef7c572/src/script.sh#L154)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Markdown broke my comment. Should _L001_ should be \_L001\_ 
</blockquote></details>

~~Should~~ L001 should be \_L001\_


<!-- Sent from Reviewable.io -->

kjwinfield committed:
check input parameters, and allow more than two lanes

kjwinfield reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: 8 of 9 files reviewed, 6 unresolved discussions (waiting on @mattgarner)

___
*[`README.md` line 11 at r14](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NH8cobO59hSdrk2dAiD:-NJp6EhZ4dYQfwG2igXv:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/bf3753907a2fd8b14f7db879dfaf4e4f0c36cf58/README.md#L11)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

Does the app check for the presence of these within the string which is passed in?
</blockquote></details>

Done.
___
*[`src/script.sh` line 133 at r17](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NJeGdwN5O3Kku57gQnt:-NJjCJJL26vt9Ohd_vS_:bb6un1j) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/6a587cbd6817a58810eeb509c20fede32ef7c572/src/script.sh#L133)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

This is not used after being defined. Remove?
</blockquote></details>

Done
___
*[`src/script.sh` line 154 at r17](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NJeGwde8yzpMtc5GbrO:-NJp7Q-U-JZGUscxrJzx:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/6a587cbd6817a58810eeb509c20fede32ef7c572/src/script.sh#L154)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

~~Should~~ L001 should be \_L001\_
</blockquote></details>

Done.
___
*[`src/script.sh` line 160 at r17](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1#-NJeHi9yDUcXqz0sd8X7:-NJp7UHV3ahCPBfKgDD0:b-896fix) ([raw file](https://github.com/eastgenomics/eggd_staraligner/blob/6a587cbd6817a58810eeb509c20fede32ef7c572/src/script.sh#L160)):*
<details><summary><i>Previously, mattgarner wrote…</i></summary><blockquote>

This will fail if we have != 2 RGs. A more generalised approach would be:

First, the reads for each RG need to be passed to `--readFilesIn`  in Lane sorted order within each R1/R2 group
like:
`X1_S1_R1_L001.fastq.gz,X1_S1_R1_L002.fastq.gz,X1_S1_R1_L003.fastq.gz X1_S1_R2_L001.fastq.gz,X1_S1_R2_L002.fastq.gz,X1_S1_R2_L003.fastq.gz`
or simplified
`L001_R1,L002_R1,L003_R1 L001_R2,L002_R2,L003_R2`

then all read groups need to be passed to `--outSAMattrRGline` in the same Lane sorted order:
`L001 , L002 , L003`
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

kjwinfield commented:
### Updates
Added ability to process more than two read groups. As seen in this job: [job-GKVKQ9042QQxJ60z6jvyXP8j]( https://platform.dnanexus.com/projects/GFfV0qQ42QQVf6VX12zxxvx3/monitor/job/GKVKQ9042QQxJ60z6jvyXP8j)

![Screenshot from 2022-12-21 14-50-21](https://user-images.githubusercontent.com/71272357/208933197-c12aa26c-eb34-47dd-94f1-c72157bc4166.png)

Added test to check none of the input parameters passed in the `--parameters` input string are set within the app. Shown in this job which failed and intentionally had an incorrect parameter as an input: [job-GKVKVK842QQqpVGj6yJq9Yp6](https://platform.dnanexus.com/projects/GFfV0qQ42QQVf6VX12zxxvx3/monitor/job/GKVKVK842QQqpVGj6yJq9Yp6)

![Screenshot from 2022-12-21 14-49-29](https://user-images.githubusercontent.com/71272357/208933000-bfb72db1-879f-4d5a-8174-f6f19cb50f99.png)

mattgarner reviewed (commented):
> Reviewed 1 of 1 files at r18, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: all files reviewed, 1 unresolved discussion (waiting on @kjwinfield)



<!-- Sent from Reviewable.io -->

mattgarner reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_staraligner/1)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @kjwinfield)



<!-- Sent from Reviewable.io -->
